sql/catalyst/src/test/scala/org/apache/spark/sql/connector/InMemoryTable.scala

```diff
@@ -51,7 +51,7 @@ class InMemoryTable(
     val distribution: Distribution = Distributions.unspecified(),
     val ordering: Array[SortOrder] = Array.empty)
   extends Table with SupportsRead with SupportsWrite with SupportsDelete
-      with SupportsMetadataColumns {
+      with SupportsMetadataColumns with TruncatableTable {
```

Comment:
I think that `SupportsDelete` should implement `TruncatableTable`, similar to how `SupportsOverwrite` implements `SupportsTruncate`. That way it isn't necessary to for implementations to support both delete and truncate separately.

Reply:
If you are sure that if an implementation supports `SupportsDelete ` then it must support `TruncatableTable` too. But I slightly doubt about it. I could imagine a case when an implementation can delete rows by a filter but cannot guarantee atomic truncation.

Reply:
Truncate is equivalent to `deleteWhere(true)`. Why would that not be equivalent?